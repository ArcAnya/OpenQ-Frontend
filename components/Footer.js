const Footer = () => {
  return (
    <div className="footer">
      <h1>Copyright 2021 OpenQ Â©</h1>
    </div>
  );
};

export default Footer;
