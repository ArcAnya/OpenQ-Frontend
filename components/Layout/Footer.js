// Third party
import React from 'react';
import Image from 'next/image';
import Link from 'next/link';

// Custom
import CopyAddressToClipboard from '../Copy/CopyAddressToClipboard';

const Footer = () => {
	const year = new Date().getFullYear();
	const iW = 16;
	const iH = 16;
	return (
		<div className="text-tinted justify-center w-full md:pl-20" style={{ position: 'absolute', bottom: '0px' }}>
			<div className='pt-2 border-t border-web-gray px-12'>
				<div className='w-full'>
					<div>These are currently unaudited contracts. OpenQ Labs assumes no responsibility or liability for any transaction errors, faults or losses which occur while minting, funding, refunding or claiming a bounty.</div>
				
				</div>
			</div>
			<div className="flex flex-wrap items-center justify-between pb-2 px-12">
				<div className='flex-0'>Copyright {year} OpenQ Â©</div>
				<div className='flex flex-wrap justify-content-end w-max text-right'>
					<div className='min-w-[100px] flex gap-4'>
						<Link href={'https://twitter.com/openqlabs'}>
							<a target={'_blank'}>
								<Image src={'/social-icons/twitter.svg'} width={iW} height={iH} />
							</a>
						</Link>
						<Link href={'https://github.com/OpenQDev/'}>
							<a target={'_blank'}>
								<Image src={'/social-icons/github-logo-white.svg'}  width={iW} height={iH}/>
							</a>
						</Link>
						<Link href={'https://discord.gg/puQVqEvVXn'}>
							<a target={'_blank'}>
								<Image src={'/social-icons/discord.svg'}  width={iW} height={iH}/>
							</a>
						</Link>
					</div>
					<CopyAddressToClipboard data={'0x06b306C85E5F33b1b2D971822ce0eD42FB7aB9a1'} styles="pt-0 w-64"/>
				</div>
			</div>
		</div>);
};

export default Footer;
