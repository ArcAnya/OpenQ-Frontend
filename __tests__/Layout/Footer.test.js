import React from 'react';
import Footer from '../../components/Layout/Footer';
import ShallowRenderer from 'react-test-renderer/shallow';
describe('Footer', () => {
  it('should match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<Footer />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
