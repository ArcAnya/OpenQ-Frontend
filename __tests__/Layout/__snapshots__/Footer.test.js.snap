// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match DOM Snapshot 1`] = `
<div
  className="flex justify-center justify-items-center full"
>
  <div
    className=" text-primary text-sm px-4 lg:px-20 max-w-[1300px] lg:py-12 py-4 xs:grid flex flex-col content-start items-start gap-4 xs:gap-y-0 lg:grid-cols-[0.5fr_1fr_1fr_1.3fr_2.5fr] grid-cols-[1fr_1fr] lg:grid-rows-2 grid-rows-4 lg:grid-flow-col xs:items-center lg:flex-row w-full justify-between xs:content-center font-semibold grid-flow-row"
  >
    <div
      className="font-semibold font-sans text-3xl"
    >
      OpenQ
    </div>
    <OpenQSocials />
    <ForwardRef(LinkComponent)
      className="text-lg lg:justify-self-center"
      href="/batch"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        Batch Mint
      </span>
    </ForwardRef(LinkComponent)>
    <div
      className="flex flex-col lg:items-center gap-4 lg:gap-0"
    >
      <ForwardRef(LinkComponent)
        className="text-lg lg:justify-self-center"
        href="/terms-of-use"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span>
          Terms of Use
        </span>
      </ForwardRef(LinkComponent)>
      <ForwardRef(LinkComponent)
        className="text-lg lg:justify-self-center"
        href="https://www.openq.dev/privacy"
        rel="noopener noreferrer"
        target="_blank"
      >
        Privacy Policy
      </ForwardRef(LinkComponent)>
    </div>
    <ForwardRef(LinkComponent)
      className="text-lg lg:justify-self-center "
      href="https://openq.canny.io/openq-feature-requests"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        Feature requests 
      </span>
    </ForwardRef(LinkComponent)>
    <ForwardRef(LinkComponent)
      className="text-lg lg:justify-self-center"
      href="https://docs.openq.dev"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span>
        Documentation
      </span>
    </ForwardRef(LinkComponent)>
    <div
      className="text-lg text-muted "
    >
      Copyright 
      2023
       
      <span
        className="whitespace-nowrap"
      >
        OpenQ Â©
      </span>
    </div>
    <div />
    <div
      className="flex flex-wrap lg:gap-2 items-center justify-content-between text-muted w-full lg:w-fit text-right py-2 "
    >
      <div
        className="min-w-[100px] flex gap-4 cursor-pointer"
        onClick={[Function]}
      >
        <span>
           Smart Contracts
        </span>
        <span
          onClick={[Function]}
        >
          <ChevronDownIcon />
        </span>
      </div>
      <div
        className="relative h-6"
      >
        <div
          className="relative w-48 undefined"
        >
          <button
            className="block"
            value={1}
          >
            <CopyAddressToClipboard
              clipping={
                [
                  5,
                  38,
                ]
              }
              styles="pt-0 w-40"
            />
          </button>
        </div>
      </div>
    </div>
    <span
      className="text-muted col-span-2 w-3/4"
    >
      Contracts currently not audited, use at your own risk.
    </span>
  </div>
  <LoadingBar />
</div>
`;
